source/cgal/surface_mesh.cc

```diff
@@ -0,0 +1,126 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+#include <deal.II/base/patterns.h>
+
+#include <deal.II/cgal/surface_mesh.h>
+
+#ifdef DEAL_II_WITH_CGAL
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace
+{
+  template <typename dealiiFace, typename Container, typename CGAL_Mesh>
+  void
+  add_facet(const dealiiFace &face,
+            const Container & deal2cgal,
+            CGAL_Mesh &       mesh,
+            const bool        clockwise_ordering = true)
+  {
+    const unsigned                                nv = face->n_vertices();
+    std::vector<typename CGAL_Mesh::Vertex_index> indices;
+
+    switch (nv)
```

Comment:
I really don't like these abbreviations, since I have to go up in the code to the definition.
```suggestion
    switch (face->n_vertices())
```

Reply:
Also, you might want to consider here to check for the reference-cell type. Makes it easier to understand the code below. Switch-statement does work with reference cells, doesn't it!?